packages/js/product-editor/src/components/block-editor/block-editor.tsx

```diff
@@ -0,0 +1,115 @@
+/**
+ * External dependencies
+ */
+import { BlockInstance } from '@wordpress/blocks';
+import { createElement, useState, useMemo } from '@wordpress/element';
+import { Product } from '@woocommerce/data';
+import { useSelect, select as WPSelect } from '@wordpress/data';
+import { uploadMedia } from '@wordpress/media-utils';
+import {
+	// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+	// @ts-ignore No types for this exist yet.
+	BlockBreadcrumb,
+	BlockEditorKeyboardShortcuts,
+	BlockEditorProvider,
+	BlockList,
+	// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+	// @ts-ignore No types for this exist yet.
+	BlockTools,
+	BlockInspector,
+	EditorSettings,
+	EditorBlockListSettings,
+	WritingFlow,
+	ObserveTyping,
+} from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import { Sidebar } from '../sidebar';
+
+type BlockEditorProps = {
+	product: Partial< Product >;
+	settings: Partial< EditorSettings & EditorBlockListSettings > | undefined;
+};
+
+export function BlockEditor( { settings: _settings }: BlockEditorProps ) {
+	const [ blocks, updateBlocks ] = useState< BlockInstance[] >();
+
+	const canUserCreateMedia = useSelect( ( select: typeof WPSelect ) => {
+		const { canUser } = select( 'core' ) as Record<
+			string,
+			( ...args: string[] ) => boolean
+		>;
+		const _canUserCreateMedia = canUser( 'create', 'media' );
```

Comment:
I find these two lines a little confusing:
```js
const _canUserCreateMedia = canUser( 'create', 'media' );
return _canUserCreateMedia || _canUserCreateMedia !== false;
```

Is that the same as just:
```js
return canUser( 'create', 'media' ) !== false;
```

Reply:
Good catch! It's done now.